### PR TITLE
fix: restart blobfuse proxy when necessary

### PR DESCRIPTION
**Pull request by cvvz**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
**Testing result:**

**uninstall old version and install new version:**
image

 
**reinstall the same driver**
image

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test
so this PR already has logic of PR: https://github.com/kubernetes-sigs/blob-csi-driver/pull/744

**Comment by cvvz** [comment]
> so this PR already has logic of PR: https://github.com/kubernetes-sigs/blob-csi-driver/pull/744
Yes, it aims to solve the issue the same as #744 and solve the [comment](https://github.com/kubernetes-sigs/blob-csi-driver/pull/744#discussion_r961385892).

File: pkg/blobfuse-proxy/init.sh [hunk 1]
```diff
@@ -33,25 +33,50 @@ then
...
-if [ ! -f "/host/usr/bin/blobfuse-proxy" ];then
+updateBlobfuseProxy="true"
+if [ -f "/host/usr/bin/blobfuse-proxy" ];then
+  old=$(sha256sum /host/usr/bin/blobfuse-proxy | awk '{print $1}')
+  new=$(sha256sum /blobfuse-proxy/blobfuse-proxy | awk '{print $1}')
+  if [ "$old" = "$new" ];then
+    updateBlobfuseProxy="false"
+    echo "no need to update blobfuse-proxy"
+  else
+    rm -rf /host/usr/bin/blobfuse-proxy
```
Comment (andyzhangx) [inline pkg/blobfuse-proxy/init.sh:44]: since it would be overwritten later, no need to do this?

File: pkg/blobfuse-proxy/init.sh [hunk 2]
```diff
@@ -33,25 +33,50 @@ then
...
-if [ ! -f "/host/usr/lib/systemd/system/blobfuse-proxy.service" ];then
+updateService="true"
+if [ -f "/host/usr/lib/systemd/system/blobfuse-proxy.service" ];then
+  old=$(sha256sum /host/usr/lib/systemd/system/blobfuse-proxy.service | awk '{print $1}')
+  new=$(sha256sum /blobfuse-proxy/blobfuse-proxy.service | awk '{print $1}')
+  if [ "$old" = "$new" ];then
+    updateService="false"
+    echo "no need to update blobfuse-proxy.service"
+  else
+    rm -rf /host/usr/lib/systemd/system/blobfuse-proxy.service
```
Comment (andyzhangx) [inline pkg/blobfuse-proxy/init.sh:63]: since it would be overwritten later, no need to do this?

File: pkg/blobfuse-proxy/init.sh (hunk 1 above)
Comment (cvvz) [reply pkg/blobfuse-proxy/init.sh]: fixed

File: pkg/blobfuse-proxy/init.sh (hunk 2 above)
Comment (cvvz) [reply pkg/blobfuse-proxy/init.sh]: fixed

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**